flashlight/distributed/reducers/CoalescingReducer.h

```diff
@@ -26,6 +26,7 @@ class Variable;
  * ``finalize`` must be called before using a given value.
  */
 class CoalescingReducer : public Reducer {
+  bool useMixedPrecision_;
```

Comment:
See my above comment. We should just dynamically choose the `ncclType` based on the `af::dtype` if the Variable to be reduced. That way we don't potentially accidentally lose precision in the reduction if we sync parameters from an fp32 module.